akka-docs/src/test/scala/docs/stream/operators/sourceorflow/GroupBy.scala

```diff
@@ -13,12 +13,12 @@ object GroupBy {
     implicit val system: ActorSystem = ???
     //#groupBy
     Source(1 to 10)
-      .groupBy(maxSubstreams = 2, _ % 2 == 0) // create two sub-streams with odd and even numbers
+      .groupBy(maxSubstreams = 2, _ % 2) // create two sub-streams with odd and even numbers
       .reduce(_ + _) // for each sub-stream, sum its elements
       .mergeSubstreams // merge back into a stream
       .runForeach(println)
-    //25
     //30
+    //25
```

Comment:
Note that this is not deterministic, because of mergeSubstreams, either order could happen, it depends on the order the stream interpreter decides to run the operators.

Reply:
That's good to know, thanks for clarifying.